Release request for mysql api version "2023-12-01-preview"

sumitkumams opened the pull request:
This PR is raised as release request for dotnet API which include the latest spec version "2023-12-01-preview" changes.

PR Includes:
1) updated "autorest.md" for mysql, and auto-generated files.
2) test recording for "AdvancedThreatProtectionSettings" feature of mysql flex server.

sumitkumams committed:
Update "autorest.md" to generate latest sdk files

sumitkumams committed:
Create Test Recording and Push to Assets

Test Recording for "AdvancedThreatProtectionSettings"

sumitkumams committed:
remove unnecessary namespaces

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.MySql](https://apiview.dev/Assemblies/Review/b80a2697e0fb4ff9b9c79951b0360c75?revisionId=af7e11203b7b4469be3271ee2af8e9f8)

sumitkumams committed:
Merge branch 'Azure:main' into sumitkuma/mysql/2023-12-01-preview

sumitkumams committed:
Merge branch 'Azure:main' into sumitkuma/mysql/2023-12-01-preview

sumitkumams committed:
Updated dot net standar 2.0 file

the file was generate after running below command:
eng\scripts\CodeChecks.ps1 -ServiceDirectory mysql

ArthurMa1978 committed:
Merge branch 'main' into mysql

ArthurMa1978 committed:
Update
